typing/subst.ml

```diff
@@ -54,8 +56,13 @@ let add_modtype id ty s = add_modtype_path (Pident id) ty s
 
 let for_saving s = { s with for_saving = true }
 
+let change_locs s loc = { s with loc = Some loc }
```

Comment:
why `change_locs` instead of `change_loc`? (As far as I can tell there is only one location.)

Also: as far as I can tell this function is only ever called on `Subst.identity` which has `loc = None`; could it maybe be made slightly more principled by becoming `with_loc`, that fails if the subtitution already has a location set?

(If I understand correctly, this `loc` parameter could have been made a parameter of `identity`, but that would make the patch much more invasive -- I count 57 uses of Subst.identity. Or is there an important reason to specify it separately?)

Reply:
> why `change_locs` instead of `change_loc`? (As far as I can tell there is only one location.)

but several items (usually)

> Also: as far as I can tell this function is only ever called on `Subst.identity` which has `loc = None`; could it maybe be made slightly more principled by becoming `with_loc`, that fails if the subtitution already has a location set?

Sounds great.

> Or is there an important reason to specify it separately?

No.